ebpfsvc/svcmain.cpp

```diff
@@ -73,33 +75,48 @@ service_install()
         SC_MANAGER_ALL_ACCESS); // full access rights
 
     if (nullptr == scmanager) {
-        return;
+        result = GetLastError();
+        goto Exit;
     }
 
-    // Create the service
+    // Create the service as LocalService.
 
     service = CreateService(
-        scmanager,                 // SCM database
-        SERVICE_NAME,              // name of service
-        SERVICE_NAME,              // service name to display
-        SERVICE_ALL_ACCESS,        // desired access
-        SERVICE_WIN32_OWN_PROCESS, // service type
-        SERVICE_DEMAND_START,      // start type
-        SERVICE_ERROR_NORMAL,      // error control type
-        path,                      // path to service's binary
-        nullptr,                   // no load ordering group
-        nullptr,                   // no tag identifier
-        nullptr,                   // no dependencies
-        nullptr,                   // LocalSystem account
-        nullptr);                  // no password
+        scmanager,                     // SCM database
+        SERVICE_NAME,                  // name of service
+        SERVICE_NAME,                  // service name to display
+        SERVICE_ALL_ACCESS,            // desired access
+        SERVICE_WIN32_OWN_PROCESS,     // service type
+        SERVICE_AUTO_START,            // start type
```

Comment:
why auto start instead of demand start like it used to be?
I don't see anything in the PR description that motivates this change.

Reply:
Yes it should be demand_start. changed it back.